File: src/lib/pjecalc/core/dominio/calculo/relatorio/irpf-atualizacao-jr-adapter.ts
/**
 * PJe-Calc v2.15.1 — IrpfAtualizacaoJRAdapter (abstract)
 * Porte TS-adaptado de:
 *   br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio.IrpfAtualizacaoJRAdapter
 *
 * Ref Java: pjecalc-fonte/.../relatorio/IrpfAtualizacaoJRAdapter.java
 */
import type Decimal from 'decimal.js';
import { JRAdapter, JRAdapterDataSource } from './jr-adapter';

export abstract class IrpfAtualizacaoJRAdapter extends JRAdapter {
  abstract getOcorrencias(): JRAdapterDataSource<IrpfAtualizacaoItemAdapter>;
  abstract getTotalDevido(): Decimal;
  abstract getTotalJuros(): Decimal;
  abstract getTotalMulta(): Decimal;
  abstract getTotalGeral(): Decimal;
  abstract getTotalPago(): Decimal;
  abstract getTotalDiferenca(): Decimal;
}

export abstract class IrpfAtualizacaoItemAdapter extends JRAdapter {
  abstract getDataOcorrencia(): Date;
  abstract getDataEvento(): Date;
  abstract getValorBase(): Decimal;
  abstract getAliquota(): Decimal;
  abstract getValorDeducao(): Decimal;
  abstract getDevido(): Decimal;
  abstract getJuros(): Decimal;
  abstract getMulta(): Decimal;
  abstract getTotal(): Decimal;
  abstract getPago(): Decimal;
  abstract getDiferenca(): Decimal;
}

File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/calculo/relatorio/IrpfAtualizacaoJRAdapter.java
/*
 * Decompiled with CFR 0.152.
 * 
 * Could not load the following classes:
 *  net.sf.jasperreports.engine.data.JRBeanCollectionDataSource
 */
package br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio;

import br.jus.trt8.pjecalc.base.comum.Periodo;
import br.jus.trt8.pjecalc.base.comum.relatorio.JRAdapter;
import java.math.BigDecimal;
import java.util.Date;
import net.sf.jasperreports.engine.data.JRBeanCollectionDataSource;

public abstract class IrpfAtualizacaoJRAdapter
extends JRAdapter {
    public abstract JRBeanCollectionDataSource getIrpfPorEvento();

    public abstract Periodo getPeriodoTotal();

    public abstract boolean isRegimeCaixa();

    public abstract class OcorrenciaDeIrpfPagamentoAdapter
    extends JRAdapter {
        public abstract BigDecimal getDevido();

        public abstract BigDecimal getTaxaJuros();

        public abstract BigDecimal getValorJuros();

        public abstract BigDecimal getTaxaMulta();

        public abstract BigDecimal getValorMulta();

        public abstract BigDecimal getTotal();

        public abstract BigDecimal getPago();

        public abstract BigDecimal getDevidoDiferenca();

        public abstract BigDecimal getTaxaJurosDiferenca();

        public abstract BigDecimal getValorJurosDiferenca();

        public abstract BigDecimal getTaxaMultaDiferenca();

        public abstract BigDecimal getValorMultaDiferenca();

        public abstract BigDecimal getTotalDiferenca();

        public abstract Date getDataEvento();

        public abstract Boolean getPagamentoNoSaldo();

        public abstract Boolean getCalculadoNoSaldo();
    }

    public abstract class OcorrenciaDeIrpfAtualizacaoAdapter
    extends JRAdapter {
        public abstract BigDecimal getVerbas();

        public abstract BigDecimal getJuros();

        public abstract String getQuantidadeMeses();

        public abstract BigDecimal getContribuicaoSocial();

        public abstract BigDecimal getPrevidenciaPrivada();

        public abstract BigDecimal getPensaoAlimenticia();

        public abstract BigDecimal getHonorarios();

        public abstract BigDecimal getDependentes();

        public abstract BigDecimal getAposentadoMaior65Anos();

        public abstract BigDecimal getBase();

        public abstract String getFaixa();

        public abstract BigDecimal getAliquota();

        public abstract BigDecimal getDeducao();

        public abstract BigDecimal getDevido();

        public abstract String getTipo();

        public abstract boolean isAnosAnteriores();

        public abstract Date getDataEvento();

        public abstract Periodo getOcorrenciaPeriodo();
    }

    public abstract class IrpfPorEventoAdapter
    extends JRAdapter {
        public abstract Date getDataEvento();

        public abstract Boolean getHasPagamento();

        public abstract JRBeanCollectionDataSource getOcorrenciasAtualizacaoDs();

        public abstract JRBeanCollectionDataSource getOcorrenciasPagamentoDs();

        public abstract BigDecimal getTotalDevidoLiquidacao();

        public abstract BigDecimal getTotalDevidoPagamento();
    }
}

